src/simsopt/_core/dofs.py

```diff
@@ -420,23 +437,26 @@ def fd_jac(self, x=None, eps=1e-7, centered=False):
                     # the size of the Jacobian.
                     jac = np.zeros((self.nvals, self.nparams))
 
-                x[j] = x0[j] - eps
+                x[j] = x0[j] - steps[j]
                 self.set(x)
                 fminus = self.f()
 
-                jac[:, j] = (fplus - fminus) / (2 * eps)
+                jac[:, j] = (fplus - fminus) / (2 * steps[j])
 
-        else:
+        elif self.differences == "forward":
             # 1-sided differences
             f0 = self.f()
             jac = np.zeros((self.nvals, self.nparams))
             for j in range(self.nparams):
                 x = np.copy(x0)
-                x[j] = x0[j] + eps
+                x[j] = x0[j] + steps[j]
                 self.set(x)
                 fplus = self.f()
 
-                jac[:, j] = (fplus - f0) / eps
+                jac[:, j] = (fplus - f0) / steps[j]
+
+        else:
+            raise ValueError("differences should be 'centered' or 'forward'")
```

Comment:
Instead of this error message, a more appropriate would be "Finite difference method <given input str> not implemented, Available methods are "centered" and "forward".